### PR TITLE
bcda-2508 Production User Guide Changes: _since parameter

### PR DESCRIPTION
**Pull request by Dave-Sukharan**
### Fixes [BCDA-2508](https://jira.cms.gov/browse/BCDA-2508)

The _since parameter has been released. The BCDA static site technical AND non-technical user guides needed to be updated for the new feature. This PR adds sections to both pages with screenshots.

### Proposed Changes
- Added a new section to the non-technical user guide between steps 4 and 5. There are now 6 steps.
- Added a new section with heading to the technical user guide titled "Filtering your Data with since"
- Added screenshots to the non-technical user guide

### Change Details

Explained how to use transactionTime, and format dates according to the FHIR standard in both versions. Ensured references to previous sections and formatting were consistent with patterns established so far.

### Security Implications

N/A

- [N/A ] new software dependencies

- [N/A ] security controls or supporting software altered

- [N/A ] new data stored or transmitted

- [N/A ] security checklist is completed for this change

- [N/A ] requires more information or team discussion to evaluate security implications

### Acceptance Validation

Screen Shot 2020-03-25 at 3 15 21 PM

Screen Shot 2020-03-25 at 3 15 46 PM

### Feedback Requested

The majority of changes discussed were accepted and changed (thank you!). Please look over screenshots and wording for any dire concerns.

### CODE REVIEW

**Comment by msnook** [comment]
Looks like this isn't set up to be merged to `master`.  Is that on purpose?

File: production/user_guide.md [hunk 1]
```diff
@@ -175,8 +176,53 @@ As shown above, in the field labeled "Resource types requested," type "Coverage.
...
+2. Start the job to acquire data from an endpoint.
+
+#### a. Pull your historical data.
+Before using `_since` for the first time, we recommend that you retrieve all historical data from the BCDA bulk data endpoints (do not use `_since`). See [Making Your First Requests for Data](https://bcda.cms.gov/production/user-guide/#making-your-first-requests-for-data) for step-by-step instructions on how to pull your historical data.
+
+Once you have retrieved your historical data, we recommend using the date of your most recent pull for `_since`. You may retrieve this date by viewing the [_transactionTime_](https://hl7.org/Fhir/uv/bulkdata/export/index.html#response---complete-status) from your last `_since` call.
+
+By retrieving your historical data before calling `_since`, this ensures that there will be no gaps in the claims data delivered from BCDA.
+
+**Note: Do not input dates before 02-12-2020 into `_since`. Limitations of the Beneficiary FHIR Data (BFD) Server prevent data before 02-12-2020 from being tagged correctly.  For more details, see the [Advanced User Guide](https://bcda.cms.gov/production/technical-user-guide/#filtering-your-data-with-_since).**
```
Comment (msnook) [inline production/user_guide.md]: Does this `note` contradict this: https://github.com/CMSgov/bcda-static-site/pull/42/files#diff-7b5449ca528afb4b448e5e149523f3dcR682

Comment (jmcconn120) [reply production/user_guide.md]: We chose to keep the non-technical version vague without use cases to encourage them to not use it before the date. We included more specific use cases in the advanced technical version. I think the use cases qualify as 'not being tagged correctly'

File: production/technical_userguide.md [hunk 2]
```diff
@@ -666,3 +666,67 @@ curl https://api.bcda.cms.gov/data/45/99dbc86f-74e5-4553-88a9-af3e718cb72b.ndjso
...
+
+For more information on `_since`, please consult the [FHIR standard on query parameters](https://hl7.org/Fhir/uv/bulkdata/export/index.html#query-parameters).
+
+### Usage
+There are a couple of helpful points to keep in mind when using `_since`.
+
+#### Before Using `_since` 
+Before using `_since` for the first time, we recommend that you retrieve all historical data. Once you have retrieved your historical data and begun using `_since`, you should use [`transactionTime`](https://hl7.org/Fhir/uv/bulkdata/export/index.html#response---complete-status) from your last bulk data request as the date for following `_since` calls.
+
+**Note: Due to limitations in the Beneficiary FHIR Data (BFD) Server, data from before 02-12-2020 is marked with the arbitrary `lastUpdated` date of 01-01-2020. If you input dates between 01-01-2020 and 02-11-2020 in the `_since` parameter, you will receive all historical data for your beneficiaries. Data loads from 02-12-2020 onwards have been marked with accurate dates.**
```
Comment (msnook) [inline production/technical_userguide.md]: This is the only time we mention `lastUpdated`.  Will readers be confused as to what `lastUpdated` is?

Comment (Dave-Sukharan) [reply production/technical_userguide.md]: I added a link on lastUpdated that will take them to the FHIR Spec's explanation. From context, I think technical users will understand that it is a parameter that selects based on dates.

File: production/user_guide.md (hunk 1 above)
Comment (msnook) [reply production/user_guide.md]: I'm worried that (1) they see contradicting usage advice on two different pages, and (2) we expose limitations in BFD by saying `not being tagged correctly`.

File: production/user_guide.md [hunk 3]
```diff
@@ -175,8 +176,53 @@ As shown above, in the field labeled "Resource types requested," type "Coverage.
...
+
+Before using `_since` for the first time, pull your historical data. Using the `_since` parameter subsequently comprises two steps:
+
+1. Input a date in the correct format.
+2. Start the job to acquire data from an endpoint.
+
+#### a. Pull your historical data.
+Before using `_since` for the first time, we recommend that you retrieve all historical data from the BCDA bulk data endpoints (do not use `_since`). See [Making Your First Requests for Data](https://bcda.cms.gov/production/user-guide/#making-your-first-requests-for-data) for step-by-step instructions on how to pull your historical data.
+
+Once you have retrieved your historical data, we recommend using the date of your most recent pull for `_since`. You may retrieve this date by viewing the [_transactionTime_](https://hl7.org/Fhir/uv/bulkdata/export/index.html#response---complete-status) from your last `_since` call.
```
Comment (msnook) [inline production/user_guide.md]: What does "from your last _since call" mean?  The `transactionTime` is included in every job status completion response, even if the call was made for historical data (i.e., not using `_since`).

File: production/user_guide.md [hunk 4]
```diff
@@ -175,8 +176,53 @@ As shown above, in the field labeled "Resource types requested," type "Coverage.
...
+
+By retrieving your historical data before calling `_since`, this ensures that there will be no gaps in the claims data delivered from BCDA.
```
Comment (msnook) [inline production/user_guide.md]: Technically, not "calling `_since`"... Maybe reword this to something like, "... before attempting to filter bulk data using `_since`..."

File: production/user_guide.md (hunk 3 above)
Comment (Dave-Sukharan) [reply production/user_guide.md]: Understood! Changed it to "your last bulk data request". Resolved.

File: production/user_guide.md (hunk 4 above)
Comment (Dave-Sukharan) [reply production/user_guide.md]: Accepted. Changed to before filtering with _since. Resolved.

File: production/technical_userguide.md [hunk 5]
```diff
@@ -666,3 +666,66 @@ curl https://api.bcda.cms.gov/data/45/99dbc86f-74e5-4553-88a9-af3e718cb72b.ndjso
 The response will be the requested data as [FHIR Coverage resources](https://www.hl7.org/fhir/coverage.html){:target="_blank"} in NDJSON format.
 
 An example of one such resource is available in the [guide to working with BCDA data](/data-guide/#sample-bcda-files).
+
+### Filtering Your Data with `_since`
+#### About `_since` 
+The `_since` parameter grants you the ability to apply a date parameter to your bulk data requests. Instead of receiving the full record of historical data every time you request data from an endpoint, you will be able to use `_since` to submit a date. BCDA will then produce claims data from the bulk data endpoints that have been loaded since the entered date.
+
+For more information on `_since`, please consult the [FHIR standard on query parameters](https://hl7.org/Fhir/uv/bulkdata/export/index.html#query-parameters).
```
Comment (msnook) [inline production/technical_userguide.md]: Can you apply a `{:target="_blank"}` to this link so it opens in a new tab (since it is not related to the `cms.gov` domain)?  There are other examples in this file that you can use as examples.

File: production/technical_userguide.md [hunk 6]
```diff
@@ -666,3 +666,66 @@ curl https://api.bcda.cms.gov/data/45/99dbc86f-74e5-4553-88a9-af3e718cb72b.ndjso
...
+
+#### Usage
+There are a couple of helpful points to keep in mind when using `_since`.
+
+#### Before Using `_since` 
+Before using `_since` for the first time, we recommend that you retrieve all historical data. Once you have retrieved your historical data and begun using `_since`, you should use [`transactionTime`](https://hl7.org/Fhir/uv/bulkdata/export/index.html#response---complete-status) from your last bulk data request as the date for following `_since` calls.
```
Comment (msnook) [inline production/technical_userguide.md]: Can you apply a `{:target="_blank"}` to this link so it opens in a new tab (since it is not related to the `cms.gov` domain)?  There are other examples in this file that you can use as examples.

File: production/technical_userguide.md [hunk 7]
```diff
@@ -666,3 +666,66 @@ curl https://api.bcda.cms.gov/data/45/99dbc86f-74e5-4553-88a9-af3e718cb72b.ndjso
...
+
+**Note: Due to limitations in the Beneficiary FHIR Data (BFD) Server, data from before 02-12-2020 is marked with the arbitrary [lastUpdated](https://www.hl7.org/fhir/search.html#lastUpdated) date of 01-01-2020. If you input dates between 01-01-2020 and 02-11-2020 in the `_since` parameter, you will receive all historical data for your beneficiaries. Data loads from 02-12-2020 onwards have been marked with accurate dates.**
```
Comment (msnook) [inline production/technical_userguide.md]: Can you apply a `{:target="_blank"}` to this link so it opens in a new tab (since it is not related to the `cms.gov` domain)?  There are other examples in this file that you can use as examples.

File: production/technical_userguide.md [hunk 8]
```diff
@@ -666,3 +666,66 @@ curl https://api.bcda.cms.gov/data/45/99dbc86f-74e5-4553-88a9-af3e718cb72b.ndjso
...
+
+#### Date and Timezone Formatting
+Dates and times submitted in `_since` must be listed in the FHIR _dateTime_ format (`YYYY-MM-DDThh:mm:ss+zz:zz`). Notice that, if you need to include a time, a timezone must also be specified (`+zz:zz`).
+
+* _Sample Date:_ February 20, 2020 12:00 PM EST
+* _dateTime Format:_ YYYY-MM-DDThh:mm:ss+zz:zz
+* _Formatted Sample:_ 2020-02-20T12:00:00.00-05:00
+
+More information about the FHIR _dateTime_ format can be found in the [FHIR Datatypes page](https://www.hl7.org/fhir/datatypes.html#dateTime).
```
Comment (msnook) [inline production/technical_userguide.md]: Can you apply a `{:target="_blank"}` to this link so it opens in a new tab (since it is not related to the `cms.gov` domain)?  There are other examples in this file that you can use as examples.

File: production/user_guide.md [hunk 9]
```diff
@@ -207,6 +207,58 @@ You will have twenty minutes before your token expires, and you will need to get
...
+2. Start the job to acquire data from an endpoint.
+3. Retrieve data via a job request.
+
+#### a. Pull your historical data.
+
+Before using `_since` for the first time, we recommend that you retrieve all historical data from the BCDA bulk data endpoints (do not use `_since`). Retrieving your historical data before before filtering bulk data with `_since` ensures that there will be no gaps in the claims data delivered from BCDA. 
+
+See [Making Your First Requests for Data](https://bcda.cms.gov/production/user-guide/#making-your-first-requests-for-data) for step-by-step instructions on how to pull your historical data.
+
+After retrieving your historical data, it may be helpful to use the date of your most recent bulk data request for subsequent data pulls using the `_since` parameter. You may retrieve this date by viewing the [_transactionTime_](https://hl7.org/Fhir/uv/bulkdata/export/index.html#response---complete-status) from your last bulk data request.
```
Comment (msnook) [inline production/user_guide.md]: Can you apply a `{:target="_blank"}` to this link so it opens in a new tab (since it is not related to the `cms.gov` domain)?  There are other examples in this file that you can use as examples.

File: production/user_guide.md [hunk 10]
```diff
@@ -207,6 +207,58 @@ You will have twenty minutes before your token expires, and you will need to get
...
+
+The example below demonstrates how to convert a date/time combination into the FHIR format.
+
+**Date and Time Example**
+
+* _Sample Date:_ February 20, 2020 12:00 PM EST
+* _dateTime Format:_ YYYY-MM-DDThh:mm:ss+zz:zz
+* _Formatted Sample:_ 2020-02-20T12:00:00.00-05:00
+
+More information about the FHIR dateTime format can be found on the [FHIR Datatypes page](https://www.hl7.org/fhir/datatypes.html#dateTime).
```
Comment (msnook) [inline production/user_guide.md]: Can you apply a `{:target="_blank"}` to this link so it opens in a new tab (since it is not related to the `cms.gov` domain)?  There are other examples in this file that you can use as examples.

File: production/technical_userguide.md [hunk 11]
```diff
@@ -666,3 +666,66 @@ curl https://api.bcda.cms.gov/data/45/99dbc86f-74e5-4553-88a9-af3e718cb72b.ndjso
...
+
+#### Usage Examples
+See the [Authentication and Authorization section](https://bcda.cms.gov/production/technical-user-guide/#authentication-and-authorization) above to obtain the API token needed before requesting data from any of the BCDA bulk data endpoints.
```
Comment (msnook) [inline production/technical_userguide.md]: We should not be linking to `https://bcda.cms.gov...` here, because then we can't test it locally.  The link should be relative, not absolute.  This should be changed to something like `/production/technical-user-guide/#authentication-and-authorization`.  There are some examples in this file that you can use as a reference.

File: production/user_guide.md [hunk 12]
```diff
@@ -207,6 +207,58 @@ You will have twenty minutes before your token expires, and you will need to get
...
+
+1. Input a date in the correct format.
+2. Start the job to acquire data from an endpoint.
+3. Retrieve data via a job request.
+
+#### a. Pull your historical data.
+
+Before using `_since` for the first time, we recommend that you retrieve all historical data from the BCDA bulk data endpoints (do not use `_since`). Retrieving your historical data before before filtering bulk data with `_since` ensures that there will be no gaps in the claims data delivered from BCDA. 
+
+See [Making Your First Requests for Data](https://bcda.cms.gov/production/user-guide/#making-your-first-requests-for-data) for step-by-step instructions on how to pull your historical data.
```
Comment (msnook) [inline production/user_guide.md]: We should not be linking to `https://bcda.cms.gov...` here, because then we can't test it locally. The link should be relative, not absolute. This should be changed to something like `/production/user-guide/#making-your-first-requests-for-data`. There are some examples in this file that you can use as a reference.

File: production/user_guide.md [hunk 13]
```diff
@@ -207,6 +207,58 @@ You will have twenty minutes before your token expires, and you will need to get
...
+
+After retrieving your historical data, it may be helpful to use the date of your most recent bulk data request for subsequent data pulls using the `_since` parameter. You may retrieve this date by viewing the [_transactionTime_](https://hl7.org/Fhir/uv/bulkdata/export/index.html#response---complete-status) from your last bulk data request.
+
+**Note: Do not input dates before 02-12-2020 into `_since`. Limitations of the Beneficiary FHIR Data (BFD) Server prevent data before 02-12-2020 from being tagged correctly.  For more details, see the [Advanced User Guide](https://bcda.cms.gov/production/technical-user-guide/#filtering-your-data-with-_since).**
```
Comment (msnook) [inline production/user_guide.md]: We should not be linking to `https://bcda.cms.gov...` here, because then we can't test it locally. The link should be relative, not absolute. This should be changed to something like `/production/technical-user-guide/#filtering-your-data-with-_since`. There are some examples in this file that you can use as a reference.

File: production/technical_userguide.md (hunk 5 above)
Comment (Dave-Sukharan) [reply production/technical_userguide.md]: Changed, resolved!
